core/src/main/java/com/linecorp/armeria/common/stream/DefaultStreamMessage.java

```diff
@@ -494,19 +501,13 @@ private void cleanup() {
         }
     }
 
-    // We directly run callbacks for event loops if we're already on the loop, which applies to the vast
-    // majority of cases.
-    private static boolean shouldRunInExecutor(@Nullable Executor executor) {
-        return executor != null &&
-               (!(executor instanceof SingleThreadEventLoop) || !((EventLoop) executor).inEventLoop());
-    }
-
     private static final class SubscriptionImpl implements Subscription {
 
         private final DefaultStreamMessage<?> publisher;
         private Subscriber<Object> subscriber;
         private final Executor executor;
         private final boolean withPooledObjects;
+        private boolean invokedOnSubscribe; // Always updated and read from an event loop thread
```

Comment:
Is this true for the case of executor == null?

Reply:
No, if `executor == null`. Removed the comment. It doesn't have to be volatile though due to the same reason we did not for `inOnNext`.